File: Kotlin/app/src/main/java/com/example/myapplication/utils/ColorConverter.kt
package com.example.myapplication.utils

import android.annotation.SuppressLint
import kotlin.math.abs
import kotlin.math.max
import kotlin.math.min
import kotlin.math.round

@SuppressLint("DefaultLocale")
class ColorConverter {
    companion object {
        fun rgbToHsv(r: Int, g: Int, b: Int): List<String> {
            val rp = r / 255.0
            val gp = g / 255.0
            val bp = b / 255.0
            val cmax = max(rp, max(gp, bp))
            val cmin = min(rp, min(gp, bp))
            val delta = cmax - cmin
            var h = 0.0
            if (delta == 0.0) {
                h = 0.0
            } else if (cmax == rp) {
                h = 60.0 * (((gp - bp) / delta) % 6.0)
            } else if (cmax == gp) {
                h = 60.0 * (((bp - rp) / delta) + 2.0)
            } else if (cmax == bp) {
                h = 60.0 * (((rp - gp) / delta) + 4.0)
            }
            var s = if (cmax == 0.0) 0 else delta / cmax
            s = s.toDouble()
            return listOf(
                String.format("%.2f", h),
                String.format("%.2f", s * 100),
                String.format("%.2f", cmax * 100)
            )
        }


        fun rgbToCmyk(r: Int, g: Int, b: Int): List<String> {
            val rp = r / 255.0
            val gp = g / 255.0
            val bp = b / 255.0
            val k = 1 - max(rp, max(gp, bp))
            if (k == 1.0) {
                return listOf("0", "0", "0", "100")
            }
            val c = (1 - rp - k) / (1 - k)
            val m = (1 - gp - k) / (1 - k)
            val y = (1 - bp - k) / (1 - k)
            return listOf(
                String.format("%.2f", c * 100).removeTrailingZeros(),
                String.format("%.2f", m * 100).removeTrailingZeros(),
                String.format("%.2f", y * 100).removeTrailingZeros(),
                String.format("%.2f", k * 100).removeTrailingZeros()
            )
        }

        fun rgbToHex(r: Int, g: Int, b: Int): String {
            var hex = r.toString(16).padStart(2, '0')
            hex += g.toString(16).padStart(2, '0')
            hex += b.toString(16).padStart(2, '0')
            return hex
        }

        fun hsvToRgb(h: Double, s: Double, v: Double): List<Int> {
            val ss = s/100
            val vs = v/100
            val c = vs * ss
            val x = c * (1 - abs((h / 60) % 2 - 1))
            val m = vs - c
            val (rp, gp, bp) = when {
                h < 60 -> Triple(c, x, 0.0)
                h < 120 -> Triple(x, c, 0.0)
                h < 180 -> Triple(0.0, c, x)
                h < 240 -> Triple(0.0, x, c)
                h < 300 -> Triple(x, 0.0, c)
                else -> Triple(c, 0.0, x)
            }
            val r = round(((rp + m) * 255)).toInt()
            val g = round(((gp + m) * 255)).toInt()
            val b = round(((bp + m) * 255)).toInt()
            return listOf(r, g, b)
        }

        fun cmykToRgb(c: Int, m: Int, y: Int, k: Int): List<Int> {
            val rp = 255 * (1 - c / 100.0) * (1 - k / 100.0)
            val gp = 255 * (1 - m / 100.0) * (1 - k / 100.0)
            val bp = 255 * (1 - y / 100.0) * (1 - k / 100.0)
            return listOf(round(rp).toInt(), round(gp).toInt(), round(bp).toInt())
        }

        private fun String.removeTrailingZeros(): String {
            return this.replace(Regex("\\.?0*$"), "")
        }
    }
}

File: Flutter/flutter_color_picker/lib/colorConverter.dart
import 'dart:math';

class ColorConverter {
  static List<String> rgbToHsv(int r, int g, int b) {
    var rp = r / 255.0;
    var gp = g / 255.0;
    var bp = b / 255.0;
    var cmax = max(rp, max(gp, bp));
    var cmin = min(rp, min(gp, bp));
    var delta = cmax - cmin;
    var h = 0.0;
    if (delta == 0.0) {
      h = 0.0;
    } else if (cmax == rp) {
      h = 60.0 * (((gp - bp) / delta) % 6.0);
    } else if (cmax == gp) {
      h = 60.0 * (((bp - rp) / delta) + 2.0);
    } else if (cmax == bp) {
      h = 60.0 * (((rp - gp) / delta) + 4.0);
    }
    var s = cmax == 0.0 ? 0.0 : delta / cmax;
    return [
      h.toStringAsFixed(2),
      (s * 100).toStringAsFixed(2),
      (cmax * 100).toStringAsFixed(2)
    ];
  }

  static List<String> rgbToCmyk(int r, int g, int b) {
    var rp = r / 255.0;
    var gp = g / 255.0;
    var bp = b / 255.0;
    var k = 1 - max(rp, max(gp, bp));
    if (k == 1.0) {
      return ["0", "0", "0", "100"];
    }
    var c = (1 - rp - k) / (1 - k);
    var m = (1 - gp - k) / (1 - k);
    var y = (1 - bp - k) / (1 - k);
    return [
      (c * 100).toStringAsFixed(2).replaceAll(RegExp(r'\.?0*$'), ''),
      (m * 100).toStringAsFixed(2).replaceAll(RegExp(r'\.?0*$'), ''),
      (y * 100).toStringAsFixed(2).replaceAll(RegExp(r'\.?0*$'), ''),
      (k * 100).toStringAsFixed(2).replaceAll(RegExp(r'\.?0*$'), '')
    ];
  }

  static String rgbToHex(int r, int g, int b) {
    var hex = r.toRadixString(16).padLeft(2, '0');
    hex += g.toRadixString(16).padLeft(2, '0');
    hex += b.toRadixString(16).padLeft(2, '0');
    return hex;
  }

  static List<int> hsvToRgb(double h, double s, double v) {
    var ss = s / 100;
    var vs = v / 100;
    var c = vs * ss;
    var x = c * (1 - ((h / 60) % 2 - 1).abs());
    var m = vs - c;
    var rp, gp, bp;
    if (h < 60) {
      rp = c;
      gp = x;
      bp = 0.0;
    } else if (h < 120) {
      rp = x;
      gp = c;
      bp = 0.0;
    } else if (h < 180) {
      rp = 0.0;
      gp = c;
      bp = x;
    } else if (h < 240) {
      rp = 0.0;
      gp = x;
      bp = c;
    } else if (h < 300) {
      rp = x;
      gp = 0.0;
      bp = c;
    } else {
      rp = c;
      gp = 0.0;
      bp = x;
    }
    var r = ((rp + m) * 255).round();
    var g = ((gp + m) * 255).round();
    var b = ((bp + m) * 255).round();
    return [r, g, b];
  }

  static List<int> cmykToRgb(int c, int m, int y, int k) {
    var rp = 255 * (1 - c / 100.0) * (1 - k / 100.0);
    var gp = 255 * (1 - m / 100.0) * (1 - k / 100.0);
    var bp = 255 * (1 - y / 100.0) * (1 - k / 100.0);
    return [rp.round(), gp.round(), bp.round()];
  }

  static List<int> hexToRgb(String hex) {
    var r = int.parse(hex.substring(0, 2), radix: 16);
    var g = int.parse(hex.substring(2, 4), radix: 16);
    var b = int.parse(hex.substring(4, 6), radix: 16);
    return [r, g, b];
  }
}